src/sagemaker_xgboost_container/algorithm_mode/train.py

```diff
@@ -92,6 +93,12 @@ def get_validated_dmatrices(train_path, validate_path, content_type, csv_weights
     train_val_dmatrix = train_dmatrix
     if combine_train_val and train_dmatrix is not None and val_dmatrix is not None:
         logging.info("Read both train and validation data into one DMatrix")
+        if train_path == validate_path or os.path.basename(train_path) == os.path.basename(validate_path):
+            logger.warning('Found same path for training and validation. This is not recommended and results may not '
+                           'be correct.')
+        else:
+            # Check if there is potential data redundancy between training and validation sets
+            check_data_redundancy(train_path, validate_path)
```

Comment:
Seems like it would make more sense to make these checks and throw these warnings right after the construction of `train_path` and `validate_path`: https://github.com/aws/sagemaker-xgboost-container/blob/b42040bf32cb8b59c444c0f92c3bef44c29ca902/src/sagemaker_xgboost_container/training.py#L56-L57

Reply:
@haixiw can you please change this too?